File: python/binary_heap/heap_init.py
from heap_down import heap_down


# heap_init should convert provided array to binary min-heap in linear time
def heap_init(heap: list[int]):
	# raise NotImplementedError('TODO')

	for i in reversed(range(len(heap)//2)):
		heap_down(heap, len(heap), i)

File: java/src/binary_heap/HeapInit.java
package binary_heap;

import static binary_heap.HeapDown.heapDown;

public class HeapInit {
	// heapInit should convert provided array to binary min-heap in linear time
	public static void heapInit(int[] arr) {
		for (int i = arr.length/2; i >=0; i--) {
			heapDown(arr, arr.length, i);
		}
	}
}
